src/network/networkd-dhcp-server.c

```diff
@@ -402,6 +412,30 @@ int dhcp4_server_configure(Link *link) {
         return 0;
 }
 
+int config_parse_dhcp_server_relay_agent_suboption(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        char **suboption_value = data;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (!startswith(rvalue, "string:"))
+                return -EINVAL;
+
+        return free_and_strdup(suboption_value, rvalue + strlen("string:"));
```

Comment:
startswith() already returns a ptr to the first char after the `:`, i.e. just use that